mantle/kola/harness.go

```diff
@@ -854,10 +864,29 @@ func runProvidedTests(testsBank map[string]*register.Test, patterns []string, mu
 			return reRunErr
 		}
 	}
+
+	// Ignore the error when only denied tests with Warn:true feaute failed
+	if runErr != nil && allTestsWarnOnError(testResults.getResults()) {
+		return nil
+	}
+
 	// If the intial run failed and the rerun passed, we still return an error
 	return runErr
 }
 
+func allTestsWarnOnError(tests []*harness.H) bool {
+	for _, test := range tests {
+		if !test.Failed() {
+			continue
+		}
+		if !IsWarningOnFailure(test.Name()) {
+			return false
+		}
+		fmt.Printf("--- ⚠️ IGNORE: \"%s\"\n", test.Name())
```

Comment:
I think we can skip this maybe. I don't think it's required but if we could it would be better to put the warning at the place the test fails:

```
--- FAIL: ext.config.files.license (25.83s)
        cluster.go:162: Error: Unit kola-runext-23.service exited with code 1
        cluster.go:162: 2023-07-21T03:30:57Z cli: Unit kola-runext-23.service exited with code 1
        harness.go:1241: kolet failed: : kolet run-test-unit failed: Process exited with status 1
```

becomes 

```
--- WARN: ext.config.files.license (26.83s)
        cluster.go:162: Error: Unit kola-runext-23.service exited with code 1
        cluster.go:162: 2023-07-21T03:30:57Z cli: Unit kola-runext-23.service exited with code 1
        harness.go:1241: kolet failed: : kolet run-test-unit failed: Process exited with status 1
```